components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,771 @@
+                    var isValidFirstElem = (startingDayOfMonthOptionArr[0] === '*' &&
+                                                        isValidateNumericValue([startingDayOfMonthOptionArr[1]], 1, 31));
+                    return isValidElements || isValidFirstElem;
+                } else if(dayOfMonth.includes('-') && dayOfWeek === '?') {
+                    var dayOfMonthRangeArr = dayOfMonth.split('-');
+
+                    return isValidateNumericValue(dayOfMonthRangeArr, 1, 31) &&
+                                    (parseInt(dayOfMonthRangeArr[0]) < parseInt(dayOfMonthRangeArr[1]));
+
+                } else if(typeof dayOfMonth === 'string' && dayOfWeek === '?') {
+                       return isValidateNumericValue([dayOfMonth], 1, 31);
+                }
+            });
+        };
+
+        var MONTH_LIST = ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct', 'nov', 'dec'];
+
+        var isValidateStringValue = function(monthArr, dataArr) {
+            errorMsg = "Invalid Cron Expression";
+            return monthArr.every(month => {
+                return dataArr.includes(month.toLowerCase());
+            })
+        };
+
+        var isValidMonthValue = function(month) {
+            if(month === '*') {
+                return true;
+            }
+            if(month.includes('_')){
+                return false;
+            }
+            if (month.search(/[^\w-,*\/]/) !== -1) {
+                return false;
+            }
+            var list = month.split(",");
+            return list.every(function (month){
+                if(month.includes('/')) {
+                    var startingDayOfMonthOptionArr = month.split('/');
+                    var isValidElements = (isValidateNumericValue([startingDayOfMonthOptionArr[0]], 1, 12) &&
+                                                        isValidateNumericValue([startingDayOfMonthOptionArr[1]], 1, 12));
+                    var isValidFirstElem = (startingDayOfMonthOptionArr[0] === '*' &&
+                                                        isValidateNumericValue([startingDayOfMonthOptionArr[1]], 1, 12));
+                    return isValidElements || isValidFirstElem;
+
+                } else if(month.includes('-')) {
+                    var monthRangeArr = month.split('-');
+                    var validMonthRange = parseInt(monthRangeArr[0]) < parseInt(monthRangeArr[1]);
+                    var validMonthStrRange = MONTH_LIST.indexOf(monthRangeArr[0]) < MONTH_LIST.indexOf(monthRangeArr[1]);
+
+                    return !isNaN(parseInt(monthRangeArr[0])) && !isNaN(parseInt(monthRangeArr[1])) ?
+                        isValidateNumericValue(monthRangeArr, 1, 12) && validMonthRange :
+                        isValidateStringValue(monthRangeArr, MONTH_LIST) && validMonthStrRange;
+                } else if(typeof month === 'string') {
```

Comment:
why you have to check the type for string ? isnt all the inputs going to be string ??

Reply:
fixed in 0fb87f3